src/utils/exchange.js

```diff
@@ -95,3 +105,46 @@ export const parseOffer = (
 export const isWethConvertedTx = (fromAssetSymbol: string, contractAddress: string): boolean => {
   return fromAssetSymbol === ETH && contractAddress === WETH[chainId].address;
 };
+
+/* eslint-disable i18next/no-literal-string */
+const setAllowanceAbiFunction = [{
+  name: 'approve',
+  outputs: [{ type: 'bool', name: 'out' }],
+  inputs: [{ type: 'address', name: '_spender' }, { type: 'uint256', name: '_value' }],
+  constant: false,
+  payable: false,
+  type: 'function',
+  gas: 38769,
+}];
+
+const ethProvider = () => getEthereumProvider(getEnv().NETWORK_PROVIDER);
+
+export const createAllowanceTx = async (
+  fromAssetAddress: string,
+  clientAddress: string,
+  contractAddress: string,
+): Promise<AllowanceTransaction | null> => {
+  if (!clientAddress) {
+    reportOrWarn('Unable to set allowance', null, 'error');
```

Comment:
Could we change this to something like `Unable to set allowance: the clientAddress was falsey`?